components/brave_wallet_ui/components/shared/add-custom-token-form/add-custom-token-form.tsx

```diff
@@ -0,0 +1,348 @@
+        token.tokenId = tokenID ? new Amount(tokenID).toHex() : ''
+        token.logo = token.logo ? token.logo : iconURL
+        token.chainId = customAssetsNetwork.chainId
+        onAddCustomAsset(token)
+        return
+      }
+      let foundToken = foundTokenInfoByContractAddress
+      foundToken.coingeckoId = coingeckoID !== '' ? coingeckoID : foundTokenInfoByContractAddress.coingeckoId
+      foundToken.logo = foundToken.logo ? foundToken.logo : iconURL
+      foundToken.chainId = customAssetsNetwork.chainId
+      onAddCustomAsset(foundToken)
+    } else {
+      const newToken: BraveWallet.BlockchainToken = {
+        contractAddress: tokenContractAddress,
+        decimals: Number(tokenDecimals),
+        isErc20: customAssetsNetwork.coin !== BraveWallet.CoinType.SOL && !tokenID,
+        isErc721: customAssetsNetwork.coin !== BraveWallet.CoinType.SOL && !!tokenID,
+        name: tokenName,
+        symbol: tokenSymbol,
+        tokenId: tokenID ? new Amount(tokenID).toHex() : '',
+        logo: iconURL,
+        visible: true,
+        coingeckoId: coingeckoID,
+        chainId: customAssetsNetwork.chainId,
+        coin: customAssetsNetwork.coin
+      }
+      onAddCustomAsset(newToken)
+    }
+    onHideForm()
+  }, [tokenContractAddress, foundTokenInfoByContractAddress, customAssetsNetwork, iconURL, tokenDecimals, tokenName, tokenSymbol, tokenID, coingeckoID])
+
+  const onToggleShowAdvancedFields = React.useCallback(() => {
+    setShowAdvancedFields(!showAdvancedFields)
+  }, [showAdvancedFields])
+
+  const onHideNetworkDropDown = React.useCallback(() => {
+    if (showNetworkDropDown) {
+      setShowNetworkDropDown(false)
+    }
+  }, [showNetworkDropDown])
+
+  const onShowNetworkDropDown = React.useCallback(() => {
+    setShowNetworkDropDown(true)
+  }, [])
+
+  const onSelectCustomNetwork = React.useCallback((network: BraveWallet.NetworkInfo) => {
+    setCustomAssetsNetwork(network)
+    onHideNetworkDropDown()
+  }, [onHideNetworkDropDown])
+
+  const onClickCancel = React.useCallback(() => {
+    resetInputFields()
+    onHideForm()
+  }, [resetInputFields, onHideForm])
+
+  // memos
+  const isDecimalDisabled = React.useMemo((): boolean => {
+    return foundTokenInfoByContractAddress?.isErc721 ?? tokenID !== ''
+  }, [foundTokenInfoByContractAddress, tokenID])
+
+  const buttonDisabled = React.useMemo((): boolean => {
+    return tokenName === '' ||
+      tokenSymbol === '' ||
+      (tokenDecimals === '0' && tokenID === '') ||
+      tokenDecimals === '' ||
+      tokenContractAddress === '' ||
+      !customAssetsNetwork ||
+      (customAssetsNetwork?.coin !== BraveWallet.CoinType.SOL &&
+        !tokenContractAddress.toLowerCase().startsWith('0x'))
+  }, [tokenName, tokenSymbol, tokenDecimals, tokenID, tokenContractAddress, customAssetsNetwork])
+
+  // effects
+  React.useEffect(() => {
+    if (!contractAddress) return
+    setTokenContractAddress(contractAddress)
+  }, [contractAddress])
```

Comment:
You dont need an effect for this. You can set the state during render, if needed